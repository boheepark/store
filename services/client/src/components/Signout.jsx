import React, {Component} from 'react';
import {Link} from 'react-router-dom';

class Signout extends Component {

  componentDidMount() {
    this.props.signout();
  };

  render() {
    return (
      <div>
        <p>You are now signed out. Click <Link to="/signin">here</Link> to sign back in.</p>
      </div>
    );
  };
};

export default Signout;
