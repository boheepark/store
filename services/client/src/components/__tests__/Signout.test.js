import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import {MemoryRouter as Router} from 'react-router-dom';
import Signout from '../Signout';


const signout = jest.fn();


test('Signout renders properly', () => {
  const wrapper = shallow(<Signout signout={signout}/>);
  const element = wrapper.find('p');
  expect(element.length).toBe(1);
  expect(element.get(0).props.children[0]).toContain('You are now signed out.');
});


test('Signout renders a snapshot properly', () => {
  const tree = renderer.create(
    <Router>
      <Signout signout={signout}/>
    </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
