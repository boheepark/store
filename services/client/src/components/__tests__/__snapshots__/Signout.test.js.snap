// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signout renders a snapshot properly 1`] = `
<div>
  <p>
    You are now signed out. Click 
    <a
      href="/signin"
      onClick={[Function]}
    >
      here
    </a>
     to sign back in.
  </p>
</div>
`;
